test-data/unit/check-class-namedtuple.test

```diff
@@ -325,7 +325,10 @@ class X(NamedTuple):
 reveal_type(X._fields)  # N: Revealed type is "Tuple[builtins.str, builtins.str]"
 reveal_type(X._field_types)  # N: Revealed type is "builtins.dict[builtins.str, Any]"
 reveal_type(X._field_defaults)  # N: Revealed type is "builtins.dict[builtins.str, Any]"
-reveal_type(X.__annotations__)  # N: Revealed type is "builtins.dict[builtins.str, Any]"
+
+# In typeshed's stub for builtins.pyi, __annotations__ is `dict[str, Any]`,
+# but it's inferred as `Mapping[str, object]` here due to the fixture we're using
+reveal_type(X.__annotations__)  # N: Revealed type is "typing.Mapping[builtins.str, builtins.object]"
 
 [builtins fixtures/dict.pyi]
```

Comment:
This test needed to be altered slightly, as with this change, mypy starts inferring the correct type of `X.__annotations__` according to the fixture that this test uses:

https://github.com/python/mypy/blob/89469ac51aff1eefcd693d87d621577370ddedb4/test-data/unit/fixtures/dict.pyi#L16-L17